avaframe/com1DFA/particleInitialisation.py

```diff
@@ -0,0 +1,246 @@
+
+    # for final configuration get neighbors and update fields
+    particles = DFAfunC.getNeighborsC(particles, dem)
+    particles, fields = DFAfunC.updateFieldsC(cfg['GENERAL'], particles, dem, fields)
+
+    fields['pfv'] = fields['FD']
+    fields['ppr'] = fields['P']
+    fields['pfd'] = fields['FV']
+
+    # save particles to file for visualisation
+    avaDir = pathlib.Path(cfg['GENERAL']['avalancheDir'])
+    outDir = avaDir / 'Outputs' / 'com1DFA' / 'particlesIni'
+    particleTools.savePartToCsv(cfg['VISUALISATION']['particleProperties'], particlesList, outDir)
+    log.info('Initialising particles finalized, total mass: %.2f, number of particles: %d' % (np.sum(particles['m']), particles['nPart']))
+
+    return particles, fields
+
+
+def resetMassPerParticle(cfg, particles, dem, relRaster):
+    """ recompute mass of particles according to their location with respect to relRaster
+
+        Parameters
+        ------------
+        cfg: configparser object
+            configuration settings
+        particles: dict
+            dictionary with particles properties
+        dem: dict
+            dictionary with info on dem
+        relRaster: np.array
+            raster of release thickness values
+
+        Returns
+        ---------
+        particles: dict
+            updated particles dictionary with new mass
+    """
+
+    ncols = dem['header']['ncols']
+    nrows = dem['header']['nrows']
+    indPartInCell = particles['indPartInCell']
+    partInCell = particles['partInCell']
+    rho = cfg['GENERAL'].getfloat('rho')
+
+    indRelY, indRelX = np.nonzero(relRaster)
+    particles['mIni'] = np.zeros(particles['nPart'])
+    for indRelx, indRely in zip(indRelX, indRelY):
+        # compute number of particles for this cell
+        hCell = relRaster[indRely, indRelx]
+        volCell = dem['areaRaster'][indRely, indRelx] * hCell
+        massCell = volCell * rho
+        ic = indRelx + ncols * indRely
+        iStart = indPartInCell[ic]
+        iStop = indPartInCell[ic+1]
+        indParts = partInCell[iStart:iStop]
+        mNew = massCell / len(indParts)
+        particles['mIni'][indParts] = mNew
+
+    return particles
+
+
+def createReleaseBuffer(cfg, inputSimLines):
+    """ add a buffer around release polygons to get boundary particles
+
+        Parameters
+        -----------
+        cfg: configparser object
+            configuration settings
+        inputSimLines: dict
+            dictionary with input data info
+
+        Returns
+        --------
+        inputSimLines: dict
+            updated inputSimLines with releaseLineBuffer
+    """
+
+    # get start indices and lengths of release polygons
+    lengthRels = inputSimLines['releaseLine']['Length']
+    startLines = inputSimLines['releaseLine']['Start']
+    count = 0
+    xBuffered = np.empty(0)
+    yBuffered = np.empty(0)
+    lengthArray = []
+    startArray = [int(startLines[0])]
+    for m in range(len(lengthRels)):
```

Comment:
m and count do the same job no? could we maybe only use the m and get rid of the count?

Reply:
good point now m is used